algorithms/schema-free/greedy_graph_coloring.gsql

```diff
@@ -0,0 +1,78 @@
+CREATE QUERY greedy_graph_coloring(SET<STRING> v_type,SET<STRING> e_type,BOOL print_color_count = TRUE,BOOL display = TRUE, STRING file_path = ""){
+  
+  /* Greedy Graph Coloring algorithm
+  This is a distributed algorithm for coloring graph on large-scale undirect graphs with     one edge type.In every iteration, the vertices send its color to its out-neighbors, and     the neighbors get assigned a different color. 
+  Conditions:
+    -  No two adjacent vertices in the graph should have same color
+    -  Until above condition is achieved , keep on finding conflicts and resolve them.
+    -  Stop the conflict detection when there are no resolution left.
+  */
+  
+  TYPEDEF tuple<INT color> testColor;
+  AndAccum<BOOL> @@isconflict=true;
+  HeapAccum<testColor>(999999,color ASC) @color;
```

Comment:
Is 999999 the maximum number of distinct colors?
Anytime we have these hardcoded limits (and GSQL doesn't allow us to parameterize them), we should call them out in a comment, so that users are away and can edit the code if they need to change the limit.

Reply:
Good Catch!! Colors can go upto potential upper limit of UINT. Will need to add that as parameter with default value as 999999